include/zephyr/bluetooth/audio/bap.h

```diff
@@ -1712,6 +1712,11 @@ struct bt_bap_unicast_client_cb {
 	 * If discovery procedure has complete both @p codec and @p ep are set to NULL.
 	 */
 	void (*discover)(struct bt_conn *conn, int err, enum bt_audio_dir dir);
+
+	/** @cond INTERNAL_HIDDEN */
```

Comment:
Hmm, if I understand you correctly you want something like 
```c
struct bt_bap_unicast_client_cb_internal {
    struct bt_bap_unicast_client_cb cb;
    sys_snode_t _node;
}
```

and then on register then we copy the callbacks from the application to `bt_bap_unicast_client_cb_internal.cb`? 

It's not a bad idea, but it has the downside that 
1) If `struct bt_bap_unicast_client_cb` is statically stored in the application (which is common for most/all Bluetooth callbacks), then we duplicate the data and thus the memory size (which is not insignificant in these larger structs). 
2) It removes the chance for applications to dynamically allocate the memory for it when it is no longer using it (after it has been unregistered). BAP unicast client callbacks cannot be unregistered at this point, but maybe in the future). 

So the idea above works OK, but it does move some memory from the application to the stack, which is not always ideal. There are some callback structs (like OTS) where certain callbacks _shall_ be non-NULL, and in that case it makes more sense to copy them around so that we can ensure correctness. 

Finally if we wanted to move some of these callbacks into flash like `BT_CONN_CB_DEFINE`, then the `node` needs to be in the public API. 

Feel free to create a GH issue for this, but there are some reasons why this approach was used here